src/markdown/generator.ml

```diff
@@ -0,0 +1,310 @@
+
+let rec source_code (s : Source.t) =
+  match s with
+  | [] -> noop
+  | h :: t -> (
+      match h with
+      | Source.Elt i -> inline i ++ source_code t
+      | Tag (None, s) -> source_code s ++ source_code t
+      | Tag (Some _, s) -> source_code s ++ source_code t)
+
+and inline (l : Inline.t) =
+  match l with
+  | [] -> noop
+  | i :: rest -> (
+      match i.desc with
+      | Text "" -> inline rest
+      | Text _ ->
+          let l, _, rest =
+            Doctree.Take.until l ~classify:(function
+              | { Inline.desc = Text s; _ } -> Accum [ s ]
+              | _ -> Stop_and_keep)
+          in
+          str {|%s|} (String.concat "" l) ++ inline rest
+      | Entity e ->
+          let x = entity e in
+          x ++ inline rest
+      | Styled (sty, content) -> style sty (inline content) ++ inline rest
+      | Linebreak -> break ++ inline rest
+      | Link (href, content) ->
+          (let rec f (content : Inline.t) =
+             match content with
+             | [] -> noop
+             | i :: rest -> (
+                 match i.desc with
+                 | Text s ->
+                     command "" (str "[%s]" s ++ str "(%s)" href) ++ f rest
+                 | _ -> noop ++ f rest)
+           in
+           f content)
+          ++ inline rest
+      | InternalLink (Resolved (link, content)) -> (
+          match link.page.parent with
+          | Some _ -> inline content ++ inline rest
+          | None ->
+              open_sq_bracket ++ inline content ++ close_sq_bracket
+              ++ open_parenthesis
+              ++ String ("#" ^ link.anchor)
+              ++ close_parenthesis ++ inline rest)
+      | InternalLink (Unresolved content) -> inline content ++ inline rest
+      | Source content -> env Any "`" "`" (source_code content) ++ inline rest
+      | Raw_markup t -> raw_markup t ++ inline rest)
+
+let rec block (l : Block.t) =
+  match l with
+  | [] -> noop
+  | b :: rest -> (
+      let continue r = if r = [] then noop else break ++ block r in
+      match b.desc with
+      | Inline i -> inline i ++ continue rest
+      | Paragraph i -> inline i ++ continue rest
+      | List (list_typ, l) ->
+          let f n b =
+            let bullet =
+              match list_typ with
+              | Unordered -> escaped "- "
+              | Ordered -> str "%d) " (n + 1)
+            in
+            bullet ++ block b ++ break
+          in
+          list ~sep:break (List.mapi f l) ++ continue rest
+      | Description _ ->
+          let descrs, _, rest =
+            Take.until l ~classify:(function
+              | { Block.desc = Description l; _ } -> Accum l
+              | _ -> Stop_and_keep)
+          in
+          let f i =
+            let key = inline i.Description.key in
+            let def = block i.Description.definition in
+            break ++ str "@" ++ key ++ str " : " ++ def ++ break ++ break
+          in
+          list ~sep:break (List.map f descrs) ++ continue rest
```

Comment:
Why not use the `definition_list` extension from pandoc ? It'll be non-standard anyway.

Reply:
Being that we now have a `—md-flavour` option, more of the renderer<`Pandoc` and `GFM`>-specific syntaxes will be added. Thanks!